app/src/main/java/com/fbada/projectone/Car.kt

```diff
@@ -0,0 +1,349 @@
+/**
+ * Enum class with available car colours
+ */
+enum class AvailableColours { Red, Blue, Green, Yellow, Silver, White, Black }
+
+/**
+ * Superclass: Car
+ * @property year year of manufacture
+ * @property colour exterior colour
+ * @property price base price
+ */
+// Superclass
+open class Car(val year: Int, val colour: AvailableColours, open var price: Double) {
+    /**
+     * Accelerates the vehicle
+     */
+    fun accelerate() {
+        println("Move forward")
+    }
+    /**
+     * Halts the vehicle
+     */
+    fun brake() {
+        println("Stopped")
+    }
+}
+
+/**
+ * Subclass BMW Manufacturer type car.  Extends Car class.
+ * @property fuelType type of fuel the car runs on
+ * @property gearBox type of gearbox
+ */
+abstract class BMW(val fuelType: String, val gearBox: String) : Car(1999, AvailableColours.Blue, 400.00) {
+    /**
+     * @property price price of the car based on fuel type and gearbox properties.
+     *  If car runs on petrol AND is automatic base price is multiplied by 30.
+     *  If car has alternative properties price is base price plus 450.
+     *
+     */
+    override var price: Double = 0.0
+        get() = if (fuelType == "Petrol" && gearBox == "Automatic") {
+            super.price * 30
+        } else {
+            super.price + 450
+        }
+    /**
+     * Reverses the vehicle
+     */
+    fun reverse() {
+        println("Move backwards")
+    }
+    /**
+     * Starts the windscreen wipers
+     */
+    fun windscreenWipers() {
+        println("Turn on windscreen wipers")
+    }
+    /**
+     * Abstract function to navigate the vehicle.
+     */
+    abstract fun navigate()
+}
+/**
+ * Subclass Mercedes Manufacturer type car.  Extends Car class.
+ * @property doors number of doors
+ * @property owners number of previous owners
+ */
+open class Mercedes(val doors: Int, val owners: Int) : Car(2010, AvailableColours.Green, 500.00) {
+    /**
+     * @property motCheck checks for valid MOT
+     */
+    val motCheck: Boolean = true
+
+    /**
+     * @property doorPremium multiplies number of doors by 1.5
+     */
+    val doorPremium: Double
+        get() = doors * 1.5  // Getter
+    /**
+     * @property price price of the car based on doors property.
+     *  If a car has less than 5 doors 10 is added to base price.
+     *  If car has 5 to 8 doors 20 is added to base price, if there are more than 8 doors 33 is added.
+     */
+    override var price: Double = 0.00 + if (doors >= 5) {
+            if (doors > 8) {
+                super.price + 33.00
+            } else super.price + 20.00
+        } else {
+            super.price + 10.00
+        }
+}
+/**
+ * Subclass Ford Manufacturer type car.  Extends Car class.
+ */
+open class Ford : Car(2020, AvailableColours.Red, 300.00) {
+    /**
+     * @property seats number of seats
+     * @property location pickup location
```

Comment:
`seats` prop has also the documentation for `location`, which shouldn't be the case.